drivers/lora/sx12xx_common.c

```diff
@@ -8,18 +8,23 @@
 #include <drivers/gpio.h>
 #include <drivers/lora.h>
 #include <logging/log.h>
+#include <sys/atomic_builtin.h>
 #include <zephyr.h>
 
 /* LoRaMac-node specific includes */
 #include <radio.h>
 
 #include "sx12xx_common.h"
 
+#define MODEM_CLAIM(data)       atomic_cas(&data.modem_usage, 0, 1)
```

Comment:
The commit message specifically mentions a semaphore. Given that the indented use is mutual exclusion, would it make more sense to implement this using vanilla mutexes and set the timeout to K_NO_WAIT?

Reply:
Mutex's can't be used as multiple calling contexts need to use the object at the same time. Namely the thread that calls the original function (`lora_send`, etc) and the system workqueue, which runs the callbacks.

Using a plain semaphore also doesn't work as you can't tell whether `k_sem_give` (the 'release' operation) succeeded or not.